### PR TITLE
Metrics-tracking Deck property

### PR DESCRIPTION
**Pull request by tsherif**
Simple Deck member prop that tracks performance statistics over a 60-frame window.

### CODE REVIEW

File: docs/api-reference/deck.md [hunk 1]
```diff
@@ -377,6 +377,13 @@ Notes:
 * The query methods offer more flexibility for developers to handle events compared to the built-in hover and click callbacks.
 
 
+## Member Variables
+
+#### metrics
+
+A map of various performance statistics for the last 60 frames of rendering.
```
Comment (Pessimistress) [inline docs/api-reference/deck.md]: Document the names of stats, and what each field means?

File: docs/api-reference/deck.md [hunk 2]
```diff
@@ -377,6 +377,27 @@ Notes:
...
+- `count`: number or amount of some metric (relevant for count statistics)
+- `average`: average time spent per operation (relevant for time statistics)
+- `hz`: number of operations per second (relevant for time statistics)
+
+Some metrics gathered in deck.gl include:
+- 'Update Attributes': time spent updating attributes
+- 'Redraw Layers': number of layers drawn
+- 'setProps Time': time spent setting properties
+- 'Redraw Count': number of times the scene was rendered
+- 'pickObject Time': time spent picking an object
```
Comment (Pessimistress) [inline docs/api-reference/deck.md]: If the stat is called `pickObject Time`, the value should be a single number that is the time. If the value is an object, the stat should just be called `pickObject`?

Comment (tsherif) [reply docs/api-reference/deck.md]: The stat names are just from grepping where stats are currently used. I agree that it's confusing, but I can't break the current API for this update.

**Comment by tsherif** [comment]
@Pessimistress this actually worked out pretty well. Decided to pull some stats in from luma as well.

File: docs/api-reference/deck.md [hunk 3]
```diff
@@ -377,6 +377,26 @@ Notes:
...
 
+## Member Variables
+
+#### metrics
+
+A map of various performance statistics for the last 60 frames of rendering. Metrics gathered in deck.gl are the following:
+- 'fps': average number of frames rendered per second
+- 'updateAttributesTime': time spent updating layer attributes
+- 'layersRedrawn': number of layers drawn
+- 'setPropsTime': time spent setting layer properties
```
Comment (Pessimistress) [inline docs/api-reference/deck.md]: This tracks the time spent in `deck.setProps` so it should be "deck properties"?

File: modules/core/src/lib/deck.js [hunk 4]
```diff
@@ -766,6 +779,63 @@ export default class Deck {
...
+      .get('CPU Time')
+      .addTime(
+        animationLoopStats.get('CPU Time') ? animationLoopStats.get('CPU Time').lastTiming : 0
+      );
+  }
+
+  /* eslint-disable complexity */
+  _getMetrics() {
+    this.metrics.fps = this.stats.get('frameRate') ? this.stats.get('frameRate').getHz() : 0;
+    this.metrics.setPropsTime = this.stats.get('setProps Time')
```
Comment (Pessimistress) [inline modules/core/src/lib/deck.js]: I think `stats.get` always returns a `Stat` instance.

File: docs/api-reference/deck.md [hunk 5]
```diff
@@ -377,6 +377,26 @@ Notes:
...
+- 'drawCount': number of times the scene was rendered
```
Comment (Pessimistress) [inline docs/api-reference/deck.md]: `framesRedrawn`?

File: modules/core/src/lib/deck.js (hunk 4 above)
Comment (tsherif) [reply modules/core/src/lib/deck.js]: You are absolutely correct. Forgot how my own API works :p

File: modules/core/src/lib/deck.js [hunk 6]
```diff
@@ -766,6 +783,40 @@ export default class Deck {
...
+    this.stats.get('CPU Time').addTime(animationLoopStats.get('CPU Time').lastTiming);
+  }
+
+  _getMetrics() {
+    this.metrics.fps = this.stats.get('frameRate').getHz();
+    this.metrics.setPropsTime = this.stats.get('setProps Time').time;
+    this.metrics.updateAttributesTime = this.stats.get('Update Attributes').time;
+    this.metrics.framesRedrawn = this.stats.get('Redraw Count').count;
+    this.metrics.pickTime =
+      this.stats.get('pickObject Time').time +
```
Comment (Pessimistress) [inline modules/core/src/lib/deck.js]: Let's just merge these three stats. I don't think anyone is using one in particular.

**Pull request merged**